### PR TITLE
Analyzer: Some PHPCS improvements

### PR DESCRIPTION
**Pull request by kraftbj**
Since I keep proposing adding code that doesn't pass PHPCS, I should do my part to reduce the excluded list :)

#### Changes proposed in this Pull Request:

* PHPCS for some files in the analyzer package.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:

* Go to '..'
*

#### Proposed changelog entry for your changes:

*

### CODE REVIEW

File: projects/packages/analyzer/scripts/jetpack-slurper.php [hunk 1]
```diff
@@ -30,8 +30,6 @@
 
 require dirname( dirname( __FILE__ ) ) . '/vendor/autoload.php';
```
Comment (anomiex) [inline projects/packages/analyzer/scripts/jetpack-slurper.php:31]: https://phabricator.wikimedia.org/T273574 strikes again.

```suggestion
require dirname( __DIR__ ) . '/vendor/autoload.php';
```

File: projects/packages/analyzer/scripts/example.php [hunk 2]
```diff
@@ -81,7 +81,6 @@
 echo "\n*** Summary of dependencies by external file\n";
```
Comment (anomiex) [inline projects/packages/analyzer/scripts/example.php:81]: Line 18 of this file has an instance of https://phabricator.wikimedia.org/T273574 too. Alas GitHub won't let me comment there, sigh.

File: projects/packages/analyzer/src/Differences/class-class-const-missing.php [hunk 3]
```diff
@@ -7,9 +7,10 @@
 
 namespace Automattic\Jetpack\Analyzer\Differences;
 
-use Automattic\Jetpack\Analyzer\PersistentList\Item as PersistentListItem;
 use Automattic\Jetpack\Analyzer\Invocations\Static_Const;
-use Automattic\Jetpack\Analyzer\Warnings\Warning; // TODO - subclasses?
+use Automattic\Jetpack\Analyzer\PersistentList\Item as PersistentListItem;
+use Automattic\Jetpack\Analyzer\Warnings\Warning;
+// TODO - subclasses?
```
Comment (anomiex) [inline projects/packages/analyzer/src/Differences/class-class-const-missing.php:13]: Feel free to move these TODO comments back up to the Warning lines. They don't make the codesniffer fail, it's just the fixer happens to move them.

**Review by anomiex — COMMENTED** [review]
Seems ok to me, other than the two places where the upstream bug causes a `dirname( __FILE__ )` to be missed and the one additional file that can be removed from the excludelist.

The other inline comment is completely optional, ignore it if you like.

File: projects/packages/analyzer/src/Differences/class-class-const-missing.php (hunk 3 above)
Comment (kraftbj) [reply projects/packages/analyzer/src/Differences/class-class-const-missing.php:13]: If the fixer moves them, I'm fine letting them lie.

**Review by anomiex — APPROVED** [review]

**Review by anomiex — COMMENTED** [review]
Something is confused, it's using the workflow from #18678 but the composer.lock from before that change.

I'm not sure, but I think it's getting the workflow from a pseudo-commit created by doing a master merge into this PR, but then we check out the PR itself rather than the pseudo-commit so phpcs-changed checks the actual changes in the PR with the side effect that it doesn't get the dependency change from #18678.

A manual merge should fix it.

**Comment by kraftbj** [comment]
Manual merge attempted!

**Pull request merged**